pkg/controllers/core/deployer_controller.go

```diff
@@ -557,6 +558,38 @@ func (r *DeployerReconciler) constructServiceForDeployer(deployer *corev1alpha1.
 	// If a service name was provided, use it, otherwise generate one based on the deployer name.
 	if deployer.Spec.ServiceName != "" {
 		service.ObjectMeta.Name = deployer.Spec.ServiceName
+
+		// Detect service name collision.
+		collidingService, err := r.getService(deployer.Namespace, deployer.Spec.ServiceName)
```

Comment:
rather than proactive detection, let the create fail. Even if this tests says the resource doesn't exist, it could be created before the controller calls create (and fails anyway)

Reply:
The idea is to provide better diagnostics for the user to take action on. Currently, apart from the service not being ready, there's no indication that the reason is because the service name was already in use. But maybe we can let the create fail and then add suitable diagnostics to the `ServiceReady` condition in this case. wdyt?

Reply:
I guess my proposal isn't bullet-proof because a service creation can fail for some reason but then someone else could immediately create a service with the desired name and we can't assume that the failure was due to the name being in use.